mmengine/runner/runner.py

```diff
@@ -2139,14 +2138,21 @@ def save_checkpoint(
             model = self.model
 
         checkpoint = {
-            'meta': meta,
-            'state_dict': weights_to_cpu(get_state_dict(model)),
-            'message_hub': self.message_hub.state_dict()
+            'meta':
+            meta,
+            'state_dict':
+            apply_to(model.state_dict(), lambda x: hasattr(x, 'cpu'),
+                     lambda x: x.cpu()),
```

Comment:
It seems like the below clause is ignored.
https://github.com/open-mmlab/mmengine/blob/fd84c210e551ebd2ac149cfcc82d443550a08f65/mmengine/runner/checkpoint.py#L629